aws/resource_aws_acm_certificate.go

```diff
@@ -38,11 +38,30 @@ func resourceAwsAcmCertificate() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+			"domain_validation_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"domain_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"validation_domain": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
 			"arn": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
-			"domain_validation_options": {
```

Comment:
This is a breaking change for anyone who implements DNS validation or otherwise uses the `domain_validation_options` attribute for outputs, etc. This PR should instead allow configuration of the existing `domain_name` and `validation_domain` child attributes of `domain_validation_options`.

Please keep it and mark it with `Deprecated: "friendly message here"` if we really are going to remove it. It would also be helpful to know why this is happening in this PR.

Reply:
I have added deprecated for the unused child attributes. The reason for removing them is because the attribute was named inappropriately for what data it contained. domain_validaton_options is an actual input parameter for acm certificates, and it is confusing to have output data in it. Therefore a new output was created, called certificate_details, which contains the same information that was removed from domain_validation_options.